hatch/app.py

```diff
@@ -46,3 +45,24 @@ async def workspace_file(workspace: str, name: str):
 
     # FastAPI supports async file responses
     return FileResponse(path)
+
+
+@app.post("/workspace/{workspace}")
+def workspace_release(
+    workspace: str,
+    release: models.Release,
+    os_user: str = Header(Required),
```

Comment:
Does this mean we have to send the HTTP header `os_user` with requests to this endpoint?

Reply:
Yes, just like job-server.

Reply:
This is to allow `osrelease` to call this api in the short term.

I plan to add the signed token auth stuff soonish, and validate() will accept either:

a) Authorization: BACKEND_TOKEN and take user from OS-User
b) Authorization: SIGNED_TOKEN and take the user from the signed token.

Huh, I just realised that `osrelease` can just signed token, so we just need to support the signed token auth, and take the user from that.

I will do that in the next PR, at which point I'll remove the need for OS-User header.

Good question!